Tests/StarknetTests/Utils/DevnetClient.swift

```diff
@@ -0,0 +1,319 @@
+import Foundation
+
+import Starknet
+
+protocol DevnetClientProtocol {
+    var gatewayUrl: String { get }
+    var feederGatewayUrl: String { get }
+    var rpcUrl: String { get }
+    
+    func start()
+    func close()
+    
+    func prefundAccount(address: Felt)
+    func deployAccount(name: String) -> DeployAccountResult
+    func deployContract(contractPath: String) -> TransactionResult
+    func declareContract(contractPath: String) -> TransactionResult
+    func readAccountDetails(accountName: String) -> AccountDetails
+}
+
+struct AccountDetails: Codable{
+    var privateKey: Felt
+    var publicKey: Felt
+    var address: Felt
+    var salt: Felt
+    
+    enum CodingKeys: String, CodingKey {
+        case privateKey = "private_key"
+        case publicKey = "public_key"
+        case address
+        case salt
+    }
+    
+    init(privateKey: Felt, publicKey: Felt, address: Felt, salt: Felt) {
+        self.privateKey = privateKey
+        self.publicKey = publicKey
+        self.address = address
+        self.salt = salt
+    }
+    
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        self.privateKey = try container.decode(Felt.self, forKey: .privateKey)
+        self.publicKey = try container.decode(Felt.self, forKey: .publicKey)
+        self.address = try container.decode(Felt.self, forKey: .address)
+        self.salt = try container.decode(Felt.self, forKey: .salt)
+        //try verifyTransactionIdentifiers(container: container, codingKeysType: CodingKeys.self)
+    }
```

Comment:
Actually, as long as we don't need public initializer, and can just use internal one, these lines can be deleted. Both initializers will work implicitly